src/ProxyClient/Varnish.php

```diff
@@ -89,6 +97,35 @@ public function setDefaultBanHeader($name, $value)
     /**
      * {@inheritdoc}
      */
+    public function invalidateTags(array $tags)
+    {
+        $tagExpression = sprintf('(%s)(,.+)?$', implode('|', array_map('preg_quote', $this->escapeTags($tags))));
+
+        return $this->ban([$this->tagsHeader => $tagExpression]);
+    }
```

Comment:
Why were these methods added back here, to the `CacheInvalidator`? Previously, we moved them out to the `TagHandler`.